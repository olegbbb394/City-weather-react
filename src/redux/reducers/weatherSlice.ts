import {createSlice, PayloadAction} from '@reduxjs/toolkit';
import {RootState} from "../store/store";
import {ICityWeatherItem} from "../../types/city-weather.model";

export interface IWeatherState {
    weatherDataArr: ICityWeatherItem[];
}

const initialState: IWeatherState = {
    weatherDataArr: [],
};

const weatherSlice = createSlice({
    name: 'weather',
    initialState,
    reducers: {
        setWeatherDataArr: (state: IWeatherState, action: PayloadAction<ICityWeatherItem[]>) => {
            state.weatherDataArr = action.payload;
        },
    },
});

export const {setWeatherDataArr} = weatherSlice.actions;
export const weatherReducer = weatherSlice.reducer;
export const selectWeatherDataArr = (state: RootState) => state.weather.weatherDataArr;