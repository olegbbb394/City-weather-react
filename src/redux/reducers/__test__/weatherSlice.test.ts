import {IWeatherState, setWeatherDataArr, weatherReducer} from "../weatherSlice";

describe('weatherSlice', () => {

    it('should render empty weatherDataArr', () => {

        const state: IWeatherState = {
            weatherDataArr: [],
        };

        const newState = weatherReducer(state, setWeatherDataArr([]));

        expect(newState.weatherDataArr).toHaveLength(0);
        expect(newState).toMatchSnapshot();
    });

    it('should render weatherDataArr with data', () => {

        const state: IWeatherState = {
            weatherDataArr: [{
                id: 703448,

                main: {
                    pressure: 1000,

                    temp: 35,
                },
                name: 'Kyiv',
                weather: [{
                    description: 'good',
                    icon: 'o2d',
                    id: 2,
                    main: 'good',
                }],

                wind: {
                    speed: 20,
                },
            }],
        };

        const newState = weatherReducer(state, setWeatherDataArr(state.weatherDataArr));

        expect(newState.weatherDataArr).toBeTruthy();
        expect(newState.weatherDataArr).toHaveLength(1);
        expect(newState).toMatchSnapshot();
    });
});