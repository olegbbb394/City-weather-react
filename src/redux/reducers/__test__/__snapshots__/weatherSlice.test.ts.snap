// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`weatherSlice should render empty weatherDataArr 1`] = `
Object {
  "weatherDataArr": Array [],
}
`;

exports[`weatherSlice should render weatherDataArr with data 1`] = `
Object {
  "weatherDataArr": Array [
    Object {
      "id": 703448,
      "main": Object {
        "pressure": 1000,
        "temp": 35,
      },
      "name": "Kyiv",
      "weather": Array [
        Object {
          "description": "good",
          "icon": "o2d",
          "id": 2,
          "main": "good",
        },
      ],
      "wind": Object {
        "speed": 20,
      },
    },
  ],
}
`;
